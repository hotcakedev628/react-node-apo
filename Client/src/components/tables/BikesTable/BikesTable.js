import React from 'react'
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const BikesTable = ({ bikes, onAddClick, onEditClick, onDeleteClick }) => (
    <div>
        <h3 className="panel-title">
            <NavLink title="Add New Bike" id="addButton" className="text-info pull-right fa fa-plus link" to='/addBike' aria-hidden="true"></NavLink>
        </h3>
        <table id="BikesTable" className="table table-hover">
            <thead>
                <tr>
                    <th>Image</th>
                    <th>Model</th>
                    <th>Color</th>
                    <th>Weight</th>
                    <th className="text-center">Update</th>
                    <th className="text-center">Delete</th>
                </tr>
            </thead>
            <tbody>
                {
                    bikes.map((item, index) =>
                        (
                            <tr key={index}>
                                <td>
                                    {item.imageName ?
                                        <img style={{ maxHeight: "60px" }} alt="bike" src={'https://s3.amazonaws.com/bike-images-22/' +item.imageName}></img>
                                        :
                                        <p>No image available</p>
                                    }
                                </td>
                                <td>{item.model}</td>
                                <td>{item.color}</td>
                                <td>{item.weight}</td>
                                <td className="text-center">
                                    <NavLink to={'/editBike/' + item._id} className="fa fa-edit text-warning" id="editButton"></NavLink>
                                </td>
                                <td className="text-center ">
                                    <a className="link" onClick={() => onDeleteClick(item)}>
                                        <i className="fa fa-trash text-danger"> </i>
                                    </a>
                                </td>
                            </tr>
                        )
                    )}
            </tbody>
        </table>
    </div>

)

export default BikesTable

BikesTable.propTypes = {
    bikes: PropTypes.array,
    onAddClick: PropTypes.func,
    onEditClick: PropTypes.func,
    onDeleteClick: PropTypes.func
};
