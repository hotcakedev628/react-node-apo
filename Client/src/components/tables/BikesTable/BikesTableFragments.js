import { NavLink } from 'react-router-dom';
import { s3Url } from '../../../config/constants';
import React from 'react'

export const BasicTableHeaders = () => (
    <React.Fragment>
        <th>Image</th>
        <th>Model</th>
        <th>Color</th>
        <th>Weight</th>
    </React.Fragment>
)

export const AdminTableHeaders = () => (
    <React.Fragment>
        <th className="text-center">Update</th>
        <th className="text-center">Delete</th>
    </React.Fragment>
)

export const BasicTableDescription = ({ item }) => (
    <React.Fragment>
        <td>
            {item.imageName ?
                <img style={{ maxHeight: "60px" }} alt="bike" src={s3Url + item.imageName}></img>
                :
                <p>No image available</p>
            }
        </td>
        <td>{item.model}</td>
        <td>{item.color}</td>
        <td>{item.weight}</td>
    </React.Fragment>
)



export const AdminTableDescriptionActions = ({ item, onDeleteClick }) => (
    <React.Fragment>
        <td className="text-center">
            <NavLink to={'/editBike/' + item._id} className="fa fa-edit text-warning" id="editButton"></NavLink>
        </td>
        <td className="text-center ">
            <a className="link" onClick={() => onDeleteClick(item)}>
                <i className="fa fa-trash text-danger"> </i>
            </a>
        </td>
    </React.Fragment>
)

