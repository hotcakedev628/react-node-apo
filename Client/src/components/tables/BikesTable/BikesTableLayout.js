import React from 'react'


export const BikesTableLayout = ({ TableHeaders, TableBodyContent }) => {
    return (
        <div className="mt-4">
            {/* <h3 className="panel-title">
            <NavLink title="Add New Bike" id="addButton" className="text-info pull-right fa fa-plus link" to='/addBike' aria-hidden="true"></NavLink>
        </h3> */}
            <table id="BikesTable" className="table table-hover">
                <thead>
                    <tr>
                        <TableHeaders />
                    </tr>
                </thead>
                <tbody>
                    <TableBodyContent />
                </tbody>
            </table>
        </div>

    )
}

