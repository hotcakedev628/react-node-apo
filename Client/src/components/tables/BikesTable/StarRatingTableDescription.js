import React from 'react'
import StarRatingComponent from 'react-star-rating-component';
import { connect } from 'react-redux';

const mapDispatchToProps = dispatch => ({ rateBike: (payload) => dispatch({ type: 'RATE_BIKE', payload }) })

export class Com extends React.Component {
    constructor(props) {
        super(props);
        this.state = { currentRate: props.rate };
    }

    render() {
        return (
            <td>
                <StarRatingComponent
                    onStarClick={(nextValue, prevValue, id) => this.props.rateBike({bikeId: id, rate:nextValue})} 
                    name={this.props.id}
                    value={this.state.currentRate || 0}
                    starCount={5}
                    onStarHover={(nextValue, prevValue, id) => this.setState({ currentRate: nextValue })}
                    onStarHoverOut={(nextValue, prevValue, id) => this.setState((oldState) => ({ currentRate: this.props.rate }))}
                />
                {
                    this.props.rate !== undefined ?
                        <p>
                            {this.props.rate} {this.props.rate >= 2 ? <span>  Stars  </span> : <span>  Star  </span>}
                        </p> :
                        <p> Not Rated yet </p>
                }
            </td>

        )
    }
}

export const StarRatingTableDescription = connect(null, mapDispatchToProps)(Com)
