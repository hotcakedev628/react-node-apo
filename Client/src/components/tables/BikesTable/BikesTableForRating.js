import React from 'react'
import { BasicTableHeaders, BasicTableDescription } from './BikesTableFragments';
import { BikesTableLayout } from './BikesTableLayout';
import StarRatingComponent from 'react-star-rating-component';


export const BikesTableForRating = ({ bikesDetails, onRateClick }) => (
    <BikesTableLayout
        TableHeaders={() =>
            <React.Fragment>
                <BasicTableHeaders />
                <th>Number of Rides</th>
                <th className="text-center">Rate</th>
            </React.Fragment>
        }

        TableBodyContent={() =>
            bikesDetails.map((item, index) =>
                (
                    <tr key={index}>
                        <BasicTableDescription item={item.bike} />
                        <td>{item.numberOfRides}</td>


                        <td className="text-center ">
                            <StarRatingComponent
                                name={item.bike._id} /* name of the radio input, it is required */
                                value={item.rate || 0 } /* number of selected icon (`0` - none, `1` - first) */
                                starCount={5} /* number of icons in rating, default `5` */
                                onStarClick={onRateClick} /* on icon click handler */
                            />
                        </td>
                    </tr>
                )
            )
        }
    />
)

