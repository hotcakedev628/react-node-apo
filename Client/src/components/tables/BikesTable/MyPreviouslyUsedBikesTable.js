import React from 'react'
import { BasicTableHeaders, BasicTableDescription } from './BikesTableFragments';
import { TableLayout } from '../TableLayout';
import { StarRatingTableDescription } from './StarRatingTableDescription';


export const MyPreviouslyUsedBikesTable = ({ bikesDetails }) => (
    <TableLayout
        TableHeaders={() =>
            <React.Fragment>
                <BasicTableHeaders />
                <th>Number of Rides</th>
                <th>My Rate</th>
            </React.Fragment>
        }

        TableBodyContent={() =>
            bikesDetails.map((item, index) =>
                (
                    <tr key={index}>
                        <BasicTableDescription item={item.bike} />
                        <td>{item.numberOfRides}</td>
                        <StarRatingTableDescription id={item.bike._id} rate={item.rate} />
                    </tr>
                )
            )
        }
    />
)

