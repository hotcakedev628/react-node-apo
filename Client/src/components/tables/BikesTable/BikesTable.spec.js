import renderer from 'react-test-renderer';
import BikesTable from './BikesTable';
import React from 'react'



describe('BasicTableDescription', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<BikesTable 
        bikes={[{model: 'as', color: 'red', _id: 'aa'}]}
        areReservationsAllowed = {true}
        onAddClick = {()=>{}}
        onEditClick = {()=>{}}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
