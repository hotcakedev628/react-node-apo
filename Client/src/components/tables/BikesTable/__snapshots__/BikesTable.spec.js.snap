// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTableDescription should match snapshot 1`] = `
<div>
  <table
    className="table table-hover"
    id="BikesTable"
  >
    <thead>
      <tr>
        <th>
          Image
        </th>
        <th>
          Model
        </th>
        <th>
          Color
        </th>
        <th>
          Weight
        </th>
        <th>
          Rate
        </th>
        <th
          className="text-center"
        >
          Reserve
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <p>
            No image available
          </p>
        </td>
        <td>
          as
        </td>
        <td>
          red
        </td>
        <td />
        <td>
          <div
            className="dv-star-rating"
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_5"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={5}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="aa_5"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_4"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={4}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="aa_4"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_3"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={3}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="aa_3"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_2"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={2}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="aa_2"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_1"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={1}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="aa_1"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
          </div>
          <p>
             Not Rated yet 
          </p>
        </td>
        <td
          className="text-center"
        >
          <a
            className="link"
            onClick={[Function]}
          >
            <i
              className="fa fa-check-square text-success"
            >
               
            </i>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
